### PR TITLE
UI cleanup, adaption for small screens, sidebar fixes

### LINKED ISSUE #55: Feature request: Allow for smaller screen sizes
Maybe just a quick finger exercise for someone who is experienced with bootstrap:
It would be wonderful if the layout were a bit more compact, so that nothing wraps in the left column and in the two wide columns of the dashboard each report fit in one line without wrapping.

### PR DESCRIPTION
**Pull request by jensb**
This is from #55 - but cleaned up a little and made easier to adapt for future changes.

I fixed the sidebar layout, added a touch sidebar (right side) for mobile small screens, made some table columns display depend on screen size, adjusted the breadcrumbs bar, decreased padding for the `card` CSS for small screens and added icons here and there. I also moved most "what to display" decisions into @media queries so changing these in batches is easier in the future.

Not all tables have been adjusted yet. I'll do the rest if you like the direction this is going.
One further idea is to also increase contrast for the `card`s, so they actually help guide the user.
Another is removing the `border-radius` for all items and going for a more material-like layout especially for small screens, that'll save another bit of space and make the overall UI even simpler. What do you think?

I did leave the top menu for medium screen sizes but you can easily hide it setting `display: none` in the the respective CSS @media(..) query. This way it's also up to every user to decide what they prefer. Positioning is not perfect yet there either.

### CODE REVIEW

File: public/style.css
Comment (antedebaas) [inline public/style.css:3]: please remove [data-bs-theme=light] as there is no light/dark theme there is just the theme. and if you want to add that these rules should apply to both

File: templates/base.html.twig
Comment (antedebaas) [inline templates/base.html.twig]: this is a reports project. DMARC is just one part of it. please remove the DMARC word

File: templates/base.html.twig
Comment (antedebaas) [inline templates/base.html.twig]: Check what? please be more descriptive to what the button does.
the previous name described exactly the action the button performs.

File: templates/dashboard/index.html.twig
Comment (antedebaas) [inline templates/dashboard/index.html.twig]: Messages not Msg

File: templates/dashboard/index.html.twig
Comment (antedebaas) [inline templates/dashboard/index.html.twig]: This project does not use the bootstrap icons but the hero-icons (https://heroicons.com/).
this should be consistent throughout the entire project so please find a icon from the hero-icons instead

File: public/style.css
Comment (antedebaas) [inline public/style.css]: the background of the breadcrums are part of the main content and should thus have the same background color

File: public/style.css
Comment (antedebaas) [inline public/style.css]: the menu should be full screen not partial screen.

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [inline templates/dmarc_reports/index.html.twig]: please use an icon from heroicons for this

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [inline templates/dmarc_reports/index.html.twig]: please use an icon from heroicons for this

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [inline templates/dmarc_reports/index.html.twig]: i had it on the right for the simple reason that its always at the same spot, even with pagination numbers added.

File: templates/domains/index.html.twig
Comment (antedebaas) [inline templates/domains/index.html.twig:65]: please use an icon from heroicons for this

File: templates/domains/index.html.twig
Comment (antedebaas) [inline templates/domains/index.html.twig:71]: please use an icon from heroicons for this

File: templates/domains/index.html.twig
Comment (antedebaas) [inline templates/domains/index.html.twig]: i had it on the right for the simple reason that its always at the same spot, even with pagination numbers added.

File: public/style.css
Comment (antedebaas) [inline public/style.css]: as i said before i want the menu items to always be in the menu.
it currently occupies the space for the title which is primary information.

also the menu had a slide open animation that is missing now

File: templates/dashboard/index.html.twig
Comment (antedebaas) [inline templates/dashboard/index.html.twig]: Id is just an internal number and holds little value. I am even thinking of removing it from public view all together. please hide on smaller screens

File: templates/dashboard/index.html.twig
Comment (antedebaas) [inline templates/dashboard/index.html.twig]: number of messages is actually not that important. its the end result that should be shown not the number of emails processed.

File: public/style.css
Comment (antedebaas) [inline public/style.css]: please revert back to padding: var(--bs-card-spacer-y) var(--bs-card-spacer-x);

File: templates/dashboard/index.html.twig
Comment (antedebaas) [inline templates/dashboard/index.html.twig:9]: please revert

File: templates/base.html.twig
Comment (antedebaas) [inline templates/base.html.twig:94]: please revert.

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [inline templates/dmarc_reports/index.html.twig:77]: please remove ... in delete entry

File: templates/domains/index.html.twig
Comment (antedebaas) [inline templates/domains/index.html.twig]: please add a space after #

File: templates/domains/index.html.twig
Comment (antedebaas) [inline templates/domains/index.html.twig]: please add a space after #

**Review by antedebaas — CHANGES_REQUESTED** [review]
see my comments.
There are definitely things in here i like. but also some that i do not.
keep in mind that not everyone had small fingers and sharp eyes and i want this project to work for everyone. not just young people.

Also before this can be merged the smtptls page should be in line with the rest so please add that too

File: public/style.css
Comment (jensb) [reply public/style.css:3]: Actually, there is. Bootstrap 5.3+ also has a dark theme which autoadjusts colors.
And if you hardcode colors they advise you to specify which theme this is for.
But we don't do this here so I'll revert.

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig]: OK

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig]: Yes, but the text was too long.
I'd suggest unifying this (and logout) with the sidebar menu, somewhat like I did for smallest screens.
It's just another menu option after all.
OK?

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig]: "Messages" force-wrapped on smaller screens and killed the layout.
But without the Id column we can probably afford it.

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig]: I did not find any reference to icons, all I could find were hardcoded.
Good to know, did you document this anywhere?

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig:9]: "xl" does not fit - at least not with the current table config. It causes line wraps in the tables.
If you want both cards side by side you need to either make the tables narrower, or use xxl.

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig]: OK.

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig]: Depends.
For my use case, "fqdn" and "organisation" is basically duplicated info so organization is not really required, but messages count is important to know because I look at reports from highest message count domains first.

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig]: OK

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig]: OK

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig]: OK, I get it - but not really. Lines may wrap so they can vertically shift.
If you really want them to be always in the same spot they must be at the top.
I centered them because I simply could not find them so well hidden in the corner. :-)

Actually, also I'd like to see some page number buttons between < and >, and a configurable "perpage" setting, can you add those?

File: templates/domains/index.html.twig
Comment (jensb) [reply templates/domains/index.html.twig]: That would wrap.
For some reason `nowrap` did not help here.
I can try again...

File: templates/domains/index.html.twig
Comment (jensb) [reply templates/domains/index.html.twig]: See above

File: templates/domains/index.html.twig
Comment (jensb) [reply templates/domains/index.html.twig:65]: ok

File: templates/domains/index.html.twig
Comment (jensb) [reply templates/domains/index.html.twig:71]: ok

File: templates/domains/index.html.twig
Comment (jensb) [reply templates/domains/index.html.twig]: See above :-)

File: public/style.css
Comment (jensb) [reply public/style.css]: I hid the medium options now, but as I said, how about leaving this option open to the user?

The animation was never visible to me (Firefox 125) so I can't check this.

File: public/style.css
Comment (jensb) [reply public/style.css]: I agree. But then the options would be at the left side, far away from the (right side) menu button.
I thought this was more ergnonomic.
How about moving the menu button to the left? That would solve both issues.

File: public/style.css
Comment (jensb) [reply public/style.css]: This uses far too much space for small screens (<768px). I tested this on two real smartphones, so it's not just an opinion. 
Initially I completely removed the `card` visual candy but you said you didn't like that, so this is the compromise. But I can directly modify the variables instead of calculating.

File: public/style.css
Comment (jensb) [reply public/style.css]: I find it hard to distinguish where the real content begins this way. 

Also, with a no-background card and with no visual clues what is what, there#s a lot of duplicated headings:
image

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig:94]: Revert what exactly?

File: templates/dashboard/index.html.twig
Comment (antedebaas) [reply templates/dashboard/index.html.twig]: "fqdn" and "organisation" are different. FQDN is receiving party and Organisation is a sending party.
as for the message count its different for each report and each report can have its own timespan too.
so a higher number here is not to be compared to a lower number in another report

File: templates/base.html.twig
Comment (antedebaas) [reply templates/base.html.twig]: ok

File: templates/dashboard/index.html.twig
Comment (antedebaas) [reply templates/dashboard/index.html.twig]: i did not (yet)

File: templates/dashboard/index.html.twig
Comment (antedebaas) [reply templates/dashboard/index.html.twig:9]: i disagree.

File: templates/base.html.twig
Comment (antedebaas) [reply templates/base.html.twig:94]: you basicly removed all the spacing making the entire design cramped and text heavy.

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig:77]: I respectfully disagree. This is basic HIG 101. 
"..." on a button or menu item indicates there will be a confirmation prompt or other dialog after clicking, which there is.
So - IMHO - the "..." belongs here, and probably in other places as well.

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig:94]: I didn't. I just moved it. The "main" tag still exists.

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig:9]: There's nothing to agree or disagree, it's just a fact which I tested. :-)
Below you can see the `col-xl-6` setting which wraps hostnames and reporters.

image

Comment (antedebaas) [reply templates/dashboard/index.html.twig:9]: That’s because you widened the menu causing the grid to be misaligned

File: public/style.css
Comment (antedebaas) [reply public/style.css]: Since English is a left to right language that would make sense as that is where your eyes start reading

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [reply templates/dmarc_reports/index.html.twig]: yeah they should be more outstanding but it wasnt a priority so far :)
it will be updated in the future

File: public/style.css
Comment (antedebaas) [reply public/style.css]: image
it works for me just fine

Comment (antedebaas) [reply public/style.css]: custom themes can have those buttons in the title bar thats fine but the primary shipped theme will not

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [reply templates/dmarc_reports/index.html.twig:77]: i've never come across this ever.
can you provide me with some sources to this?

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig:9]: This was necessary because originally the menu items wrapped and were misaligned.
Choose your poison ... :-)

I reduced the sidebar to 160px and removed the Messages column, now it works with xl too.
But long Reporter names or Domains may still wrap.

File: templates/dashboard/index.html.twig
Comment (jensb) [reply templates/dashboard/index.html.twig]: OK, agreed. Also, the only real relevant info is whether there are any failures.
That would also solve the wrapping issue above, then we can use col-xl-6 without issues.

File: templates/base.html.twig
Comment (antedebaas) [reply templates/base.html.twig:94]: what im referring to is 
image
that i want to keep this spacing

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig]: How about top right? Easier accessibility, saves screen space, fixed position, and room for a page number dropdown if so later desired.
image

File: public/style.css
Comment (jensb) [reply public/style.css]: They are not in the top bar any more. Actually, except for the menu icon and the (just decorative) title "Reports", the top menu bar is now completely without function. We could also remove it completely. What do you think?

File: templates/base.html.twig
Comment (antedebaas) [inline templates/base.html.twig:59]: Check mail (lower case m)

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig:94]: OK for >=xl (which was the only case before I started).

File: templates/base.html.twig
Comment (antedebaas) [inline templates/base.html.twig:56]: personally i'd prefeer the outlined - inbox-arrow-down icon from the heroicons set

File: public/style.css
Comment (antedebaas) [reply public/style.css]: the breadcrubs are a main navigator and are always placed on top of the content. it allows for easy navigating back

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [reply templates/dmarc_reports/index.html.twig]: please no.
leave it where it was.

you have the tendency to stick everything on top of each other. spacing makes for a relaxed view so please leave that in.
if you want raw data as much as possible you should use a database tool and look directly at the database.
this project aims to present it in a clear manner

File: public/style.css
Comment (antedebaas) [reply public/style.css]: since all the items have been moved out i think thats fine.
but place the title "Reports" on top of the menu

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig:77]: Sure!
I was going to send you a google link but this one https://stackoverflow.com/questions/637683 is even better since it directly links to Apple, KDE, Microsoft and some other HIG sources which - for once - surprisingly all agree on this. :-)

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig:59]: Fine by me.

File: templates/base.html.twig
Comment (jensb) [reply templates/base.html.twig:56]: Fine by me.

File: templates/dmarc_reports/index.html.twig
Comment (antedebaas) [reply templates/dmarc_reports/index.html.twig:77]: > The crucial factor is whether the menu option requires additional information (input or a selection) before it carries out the operation. So Help-About doesn't require an ellipsis

this is just an conformation dialog. No additional information is needed.

**Comment by antedebaas** [comment]
@jensb friendly ping

**Comment by jensb** [comment]
Hi,
sorry for the silence. Family life :-)
What is still to be done?
AFAIK all issues were resolved.

**Comment by antedebaas** [comment]
please process all comments first

**Comment by jensb** [comment]
Will have a look tonight. Sorry for the delay.

File: public/style.css
Comment (jensb) [reply public/style.css]: OK, it's up to you. Feel free to revert this if this is the only change that blocks this PR.

File: templates/dmarc_reports/index.html.twig
Comment (jensb) [reply templates/dmarc_reports/index.html.twig]: I didn't commit this, so fine by me.

File: public/style.css
Comment (jensb) [reply public/style.css]: Not entirely, the burger menu is still there. 
I will postpone this for another PR to get this one merged.

**Comment by jensb** [comment]
Done. OK like this?

**Comment by antedebaas** [comment]
i'm sorry but im never gonna merge this as is.
the design is incomplete and more important inconsistent between pages.
most of the comments i made have not been resolved.

**Comment by jensb** [comment]
I don't get it.
On my side of Github, most of the commits show up as "resolved" and my comment was the last one on each.
Most of the things you wanted were implemented and this took me quite some time.
Don't get me wrong, but this is a bit disheartening.
If there's a misunderstanding on my part (perhaps the way PRs work) please help me find it.

**Comment by antedebaas** [comment]
image
image
image
image

**Comment by antedebaas** [comment]
the design is all over the place.
its inconsistent, uses different colors for elements that should have the same color.
icons dont align etc.
its cramped and text heavy where i try to find i middle ground.

**Comment by jensb** [comment]
Hm...

Your first screenshot seems not up to date. The reduced padding only appears when the sidebar is also not there:
grafik
vs.
grafik

There's some stuff I did not commit because you didn't like it (back/fwd buttons for one), I'll check over the weekend if I missed a commit.

Regarding second screenshot, I removed the h:m:s time strings from the dates if they are 00:00:00 or 23:59 to declutter. See here: 
grafik

We can also remove time specifications totally and just keep the dates. In your original design they were always there (IIRC) so this was much more cramped.

Regarding no. 3, I don't have SMTP-TLS reports to play with so I never touched that side. This is basically your design. I did say I'd be happy to copy over the DMARC template's design as soon as you are happy with it, but not before, to avoid duplicate work.

Regarding no. 4, same thing. Me left, you right. As soon as you're happy with the left side, I can do the same for SMTP-TLS.

What else do you consider inconsistent?
Which icons specifically - except for the SMTP-TLS template which I left mostly alone - do not line up?

**Pull request closed without merging**
